### DISCUSSION TITLE
Using smithay with slint

### DISCUSSION THREAD

**Discussion by Arthex1** [slint-ui/slint#3119] [Q&A]
Hello, i have recently started using Slint with smithay for layer shell purposes, and have had a few issues. 

1. What do I pass in the "key pressed/release" window event? it takes a string, but no idea what that string is supposed to be.
2. How do popups work? Even though I properly dispatch mouse events through `MinimalSoftwareWindow::dispatch_event`, popups don't work. Why is that? Is there something special I have to do?

The code can be found [here](https://github.com/Amulet9/slint_layer_shell/blob/main/src/slint_window.rs#L239)

Slint code used:
```slint
export component MainUi inherits Window { 
    Text {
        text: "foo";
        font-size: 200px;
    }

    

    popup := PopupWindow {
        Rectangle { height:100%; width: 100%; background: yellow; }
        x: 60px; y: 60px; height: 60px; width: 50px;
    }
    TouchArea {
        height: parent.height;
        width: parent.width;
        clicked => {
            popup.show()
        }
    }
    
}
```

**Comment by tronical** [comment]
That's a little odd, you shouldn't have to do anything special for Popups. In fact, the code above looks good and it works for me locally with the software renderer (haven't tried with your code base yet).

If you add `            debug("mouse pressed at \{self.mouse-x/1px}/\{self.mouse-y/1px}");` into your `clicked` handler, do you see the same coordinates as you've sent in?

How would I go ahead and try out your code?

**Reply by Arthex1** [reply]
Apologies for the late reply, as I was on a trip. But I have updated the repository to be used stand-alone. So you can just clone the [repo](https://github.com/Amulet9/slint_layer_shell/tree/main), make a new cargo project, add the repo as a local dependency, and run the example provided in the repo. I am using Sway if that matters.

TouchArea seems to work fine, just that PopupWindow doesn't show up.
![image](https://github.com/slint-ui/slint/assets/87478573/946bbaee-5229-44cb-881b-74e965123f7c)
